src/ui/PackageItem.tsx

```diff
@@ -0,0 +1,157 @@
+import { SemVer } from 'semver';
+import { Markdown } from './Markdown.js';
+import { VersionSelector } from './VersionSelector.js';
+import { DependencyErrorSection } from './DependencyErrorSection.js';
+import { Package, ReleaseType } from './types.js';
+
+type PackageItemProps = {
+  pkg: Package;
+  selections: Record<string, string>;
+  versionErrors: Record<string, string>;
+  packageDependencyErrors: Record<
+    string,
+    {
+      missingDependentNames: string[];
+      missingDependencies: string[];
+    }
+  >;
+  loadingChangelogs: Record<string, boolean>;
+  changelogs: Record<string, string>;
+  onSelectionChange: (packageName: string, value: ReleaseType | '') => void;
+  onCustomVersionChange: (packageName: string, version: string) => void;
+  onFetchChangelog: (packageName: string) => Promise<void>;
+  setSelections: React.Dispatch<React.SetStateAction<Record<string, string>>>;
+  setChangelogs: React.Dispatch<React.SetStateAction<Record<string, string>>>;
+  isSelected: boolean;
+  onToggleSelect: () => void;
+};
+
+export function PackageItem({
+  pkg,
+  selections,
+  versionErrors,
+  packageDependencyErrors,
+  loadingChangelogs,
+  changelogs,
+  onSelectionChange,
+  onCustomVersionChange,
+  onFetchChangelog,
+  setSelections,
+  setChangelogs,
+  isSelected,
+  onToggleSelect,
+}: PackageItemProps) {
+  return (
+    <div
+      key={pkg.name}
+      id={`package-${pkg.name}`}
+      className={`border p-4 rounded-lg ${
+        selections[pkg.name] && selections[pkg.name] !== 'intentionally-skip'
+          ? 'border-gray-500'
```

Comment:
Maybe we add `border-2` in addition to or instead of a darkened border so it's a little more clear what's selected?